Earthfile

```diff
@@ -375,20 +381,58 @@ test-rust:
     BUILD +test-dataflow-jit --RUST_TOOLCHAIN=$RUST_TOOLCHAIN --RUST_BUILD_PROFILE=$RUST_BUILD_PROFILE
     BUILD +test-manager --RUST_TOOLCHAIN=$RUST_TOOLCHAIN --RUST_BUILD_PROFILE=$RUST_BUILD_PROFILE
 
+# TODO: the following two container tasks duplicate work that we otherwise do in the Dockerfile,
```

Comment:
So I guess this becomes our primary dockerfile, since containers built by CI get uploaded to the container repo, right?  Are the actual dockerfiles still used for anything?

Reply:
No it is not the primary dockerfile, the other one still is. `SAVE IMAGE` does not push containers to the repo unless we do `SAVE IMAGE --push` and invoke earthly with `earthly --push ...`.

I'm not yet comfortable using our test workflow here to construct the final docker image, given the fairly intricate sequence of steps with constructing dummy files etc for build caching. The final docker images would need to be built in release mode.

I'm adding another workflow that will run using the primary Dockerfile but only on the main branch.

Reply:
Understood.